services/cd-service/pkg/repository/repository.go

```diff
@@ -523,33 +523,66 @@ func (r *repository) ProcessQueue(ctx context.Context, health *setup.HealthRepor
 	}
 }
 
-func (r *repository) applyTransformerBatches(transformerBatches []transformerBatch, allowFetchAndReset bool, transaction *sql.Tx) ([]transformerBatch, error, *TransformerResult) {
+func (r *repository) applyTransformerBatches(transformerBatches []transformerBatch, allowFetchAndReset bool) ([]transformerBatch, error, *TransformerResult) {
 	//exhaustruct:ignore
 	var changes = &TransformerResult{}
+	transactions := make([]*sql.Tx, len(transformerBatches))
+	absoluteIndex := -1
+	if r.DB.ShouldUseEslTable() {
+		defer func(txs []*sql.Tx) { //Makes sure we always attempt a rollback on every created tx.
+			for _, tx := range txs {
+				_ = tx.Rollback() //If tx has already been closed, we ignore the error saying tx has ended. If not, we roll it back.
+			}
+		}(transactions)
+	}
+
 	for i := 0; i < len(transformerBatches); {
+		var transaction *sql.Tx
+		var txErr error
 		e := transformerBatches[i]
+		absoluteIndex++
```

Comment:
I think the `absoluteIndex` and `i` are always the same.
=> remove absoluteIndex

Reply:
ahh, I see now, nevermind